File: backend/app/schemas/explorar.py
from pydantic import BaseModel
from typing import Optional


class Attraction(BaseModel):
    id: str
    city: str
    category: str
    name: str
    description: str
    imageUrl: str
    location: str
    hours: str
    price: str
    longDescription: Optional[str] = None
    tripAdvisorUrl: Optional[str] = None
    reviews: Optional[list[dict]] = None


class DestinoPopular(BaseModel):
    city: str
    country: str
    description: str
    imageUrl: str
    categories: list[str]


class RespostaDestinos(BaseModel):
    cities: list[str]
    categories: list[str]
    destinos: list[DestinoPopular]


class RespostaAtracoes(BaseModel):
    city: str
    atracoes: list[Attraction]

File: frontend/lib/types/explorar.ts
// Espelha exatamente o schema Pydantic do backend (schemas/explorar.py)
// e a interface Attraction do mockAtracoes.ts para garantir compatibilidade.

export type City = "Rio de Janeiro" | "Paris" | "Tóquio" | "Nova York";
export type Category =
  | "Todas"
  | "Cultura e História"
  | "Natureza e Aventura"
  | "Gastronomia"
  | "Esportes";

export interface Attraction {
  id: string;
  city: string;
  category: Category;
  name: string;
  description: string;
  imageUrl: string;
  location: string;
  hours: string;
  price: string;
  longDescription?: string;
  tripAdvisorUrl?: string;
  reviews?: {
    author: string;
    text: string;
    rating: number;
  }[];
}

export interface DestinoPopular {
  city: string;
  country: string;
  description: string;
  imageUrl: string;
  categories: string[];
}

export interface RespostaDestinos {
  cities: string[];
  categories: string[];
  destinos: DestinoPopular[];
}

export interface RespostaAtracoes {
  city: string;
  atracoes: Attraction[];
}